File: Section 3/03_MVVM_Retrofit_API/PostViewModel.java
package com.mzu.mvvmretrofit.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;
import com.mzu.mvvmretrofit.model.Post;
import com.mzu.mvvmretrofit.repository.PostRepository;
import java.util.List;

public class PostViewModel extends ViewModel {
    private PostRepository repository;
    private LiveData<List<Post>> posts;

    public PostViewModel() {
        repository = new PostRepository();
        posts = repository.getPosts();
    }

    public LiveData<List<Post>> getPosts() {
        return posts;
    }
}

File: Section 3/04_Coroutines_MVVM_Retrofit_API_Kotlin/PostViewModel.kt
package com.mzu.coroutinemvvm.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.mzu.coroutinemvvm.model.Post
import com.mzu.coroutinemvvm.repository.PostRepository
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.launch

class PostViewModel : ViewModel() {
    private val repository = PostRepository()
    private val _posts = MutableStateFlow<List<Post>>(emptyList())
    val posts: StateFlow<List<Post>> = _posts

    fun fetchPosts() {
        viewModelScope.launch {
            repository.getPosts().collect { postList ->
                _posts.value = postList
            }
        }
    }
}
